lib/collection/variable.js

```diff
@@ -198,15 +237,64 @@ _.assign(Variable, /** @lends Variable */ {
          */
         'number': Number,
 
+        /**
+         * A "json" type value stores JSON data format
+         */
+        'json': {
+            /**
+             * @param {Object|Array} val
+             * @returns {String}
+             */
+            in: function (val) {
+                try {
+                    // @todo: shoud we check if `val` is a valid JSON string?
+                    val = typeof val === STRING ? val : JSON.stringify(val);
```

Comment:
is this done to accept serialised data? if it is to give polymorphic input to user... is that good?

Reply:
This is done because otherwise JSON strings will be stringified again and stored. We would try to parse it and it would fail always.

So not doing this is more work while setting and getting, with no benefit.